flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskStateManager.java

```diff
@@ -60,4 +62,9 @@ void reportTaskStateSnapshots(
 	 * @return previous state for the operator. Null if no previous state exists.
 	 */
 	OperatorSubtaskState operatorStates(OperatorID operatorID);
+
+	/**
+	 * Returns the base directory for all file-based local state of the owning subtask.
+	 */
+	File getSubtaskLocalStateBaseDirectory();
```

Comment:
I guess there will be other non file-based local states, right? What would this method return for these state objects?

Reply:
This is the manager, not the state objects. So for local recovery that is not based on local files, the backend will just not care about directories and no invoke this method.